Client/Frontend/Browser/Tabs/TabTrayViewController.swift

```diff
@@ -0,0 +1,191 @@
+        label.layer.cornerRadius = TabsButtonUX.CornerRadius
+        label.textAlignment = .center
+        label.text = String(tabManager.normalTabs.count)
+        return label
+    }()
+
+    lazy var navigationMenu: UISegmentedControl = {
+        let navigationMenu = UISegmentedControl(items: [UIImage(named: "nav-tabcounter")!.overlayWith(image: countLabel), UIImage(named: "smallPrivateMask")!, UIImage(named: "synced_devices")!])
+        navigationMenu.accessibilityIdentifier = "navBarTabTray"
+        navigationMenu.selectedSegmentIndex = tabManager.selectedTab?.isPrivate ?? false ? 1 : 0
+        navigationMenu.addTarget(self, action: #selector(panelChanged), for: .valueChanged)
+        return navigationMenu
+    }()
+
+    lazy var navigationToolbar: UIToolbar = {
+        let toolbar = UIToolbar()
+        toolbar.delegate = self
+        toolbar.setItems([UIBarButtonItem(customView: navigationMenu)], animated: false)
+        return toolbar
+    }()
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+
+    init(tabTrayDelegate: TabTrayDelegate? = nil, profile: Profile, showChronTabs: Bool = false) {
+        self.profile = profile
+        self.tabManager = BrowserViewController.foregroundBVC().tabManager
+
+        if showChronTabs {
+            self.tabTrayView = ChronologicalTabsViewController(tabTrayDelegate: tabTrayDelegate, profile: profile)
+        } else {
+            self.tabTrayView = GridTabViewController(tabManager: self.tabManager, profile: profile, tabTrayDelegate: tabTrayDelegate)
+        }
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewSetup()
+        applyTheme()
+        setupNotifications()
+    }
+
+    private func viewSetup() {
+        if let window = (UIApplication.shared.delegate?.window)! as UIWindow? {
+            window.backgroundColor = .black
+        }
+
+        navigationController?.navigationBar.shadowImage = UIImage()
+
+        // Add Subviews
+        if UIDevice.current.userInterfaceIdiom == .phone {
+            view.addSubview(navigationToolbar)
+        } else {
+            view.addSubview(navigationToolbar)
+        }
+
+        navigationToolbar.snp.makeConstraints { make in
+            make.left.right.equalTo(view)
+            make.top.equalTo(view.safeArea.top)
+        }
+
+        navigationMenu.snp.makeConstraints { make in
+            make.width.lessThanOrEqualTo(343)
+            make.height.equalTo(ChronologicalTabsControllerUX.navigationMenuHeight)
+        }
+
+        showPanel(tabTrayView)
+    }
+
+    private func setupNotifications() {
+        NotificationCenter.default.addObserver(self, selector: #selector(applyTheme), name: .DisplayThemeChanged, object: nil)
+    }
```

Comment:
since we are adding notification we should also add a `deinit` method in this class to remove any notifications that are no longer used

`
    deinit {
        NotificationCenter.default.removeObserver(self)
    }
`